qiskit_experiments/characterization/spectroscopy.py

```diff
@@ -0,0 +1,408 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Spectroscopy experiment class."""
+
+from typing import List, Optional, Tuple, Union
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Gate, Parameter
+from qiskit.exceptions import QiskitError
+import qiskit.pulse as pulse
+from qiskit.qobj.utils import MeasLevel
+from qiskit_experiments.analysis.curve_fitting import curve_fit
+from qiskit_experiments.base_analysis import BaseAnalysis
+from qiskit_experiments.base_experiment import BaseExperiment
+from qiskit_experiments import AnalysisResult
+from qiskit_experiments import ExperimentData
+from qiskit_experiments.data_processing.data_processor import DataProcessor
+from qiskit_experiments.data_processing.nodes import SVD, AverageData
+from qiskit_experiments.data_processing.nodes import Probability
+from qiskit_experiments.analysis import plotting
+
+
+class SpectroscopyAnalysis(BaseAnalysis):
+    """A class to analyze a spectroscopy experiment.
+
+    Analyze a spectroscopy experiment by fitting the data to a Gaussian function.
+    The fit function is:
+
+    .. math::
+
+        a * exp(-(x-x0)**2/(2*sigma**2)) + b
+
+    Here, :math:`x` is the frequency. The analysis loops over the initial guesses
+    of the width parameter :math:`sigma`. The measured y-data will be rescaled to
+    the interval (0,1).
+    """
+
+    # pylint: disable=arguments-differ, unused-argument
+    def _run_analysis(
+        self,
+        experiment_data: ExperimentData,
+        data_processor: Optional[callable] = None,
+        meas_level: int = MeasLevel.KERNELED,
+        meas_return: int = "single",
```

Comment:
These properties are not necessary because this is already involved in the circuit result header. I think this these information should be consumed by data processor rather than analysis class, and we need to update free-form dict of circuit result. See https://github.com/Qiskit/qiskit-experiments/issues/64 for details (If you want merge this first, this can be future extension).

Reply:
See here: 717dce0. Any upgrades to this should be left to a future PR.